pkg/metrics/client_metrics.go

```diff
@@ -31,4 +32,4 @@ func DisableRESTClientMetrics() {
 
 type noopLatency struct{}
 
-func (noopLatency) Observe(string, url.URL, time.Duration) {}
+func (noopLatency) Observe(context.Context, string, url.URL, time.Duration) {}
```

Comment:
Why are we adding `context` here? Looking at the definition of clientmetrics.RequestLatency, it doesn't appear to expect a context.

Reply:
Looks like this is part of k8s v1.21 api changes: 
https://github.com/kubernetes/client-go/blob/release-1.21/tools/metrics/metrics.go#L117
https://github.com/kubernetes-sigs/controller-runtime/blob/v0.9.2/pkg/metrics/client_go_adapter.go#L166

Reply:
ah, might have been looking at an older version of the code. LGTM